### PR TITLE
EVG-5809 Remove special compile stage handling

### PR DESCRIPTION
**Pull request by bsamek**

### CODE REVIEW

File: model/build/build.go [hunk 1]
```diff
@@ -81,7 +81,7 @@ func (b *Build) IsFinished() bool {
 // returns boolean to indicate if tasks are complete, string with either
```
Comment (ybrill) [inline model/build/build.go]: we should fix the comment to reflect the compile finished case.

**Comment by bsamek** [comment]
> What was the original motivation to treat compile tasks differently?

Evergreen was originally written to only support the server. It has two tasks, compile and push, which were special. All test tasks depended on compile, and push depended on test tasks. Now these can be handled with dependencies, and it's no longer the case that all tasks depend on compile. There are also other projects which may name a task compile and not intend it to have the same behavior as the server.

File: model/build/build.go (hunk 1 above)
Comment (bsamek) [reply model/build/build.go]: Fixed.

**Comment by ybrill** [comment]
> Now these can be handled with dependencies

So it's no longer the case that server relies on compile being treated differently?

**Comment by bsamek** [comment]
> > Now these can be handled with dependencies
> 
> So it's no longer the case that server relies on compile being treated differently?

I believe that to be true.

**Pull request merged**